### PR TITLE
[RW-5264, RW-5266][risk=no] Reporting PoC payload objects and service

### PR DESCRIPTION
**Pull request by jaycarlton**
Initial classes and services for reporting. Highlights:
* A ReportingService class responsible for gathering the information and building a payload for BigQuery
* Value classes (generated for convenience in Swagger) for the columns we need. Currently just researcher adn workspce tables (2-3 columns each)
* a method to capture a timestamp to use as a partition key (potentially)
* endpoint for kicking off the upload, to serve as cron target
* unit test class

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test:local`
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/resources/workbench-api.yaml [hunk 1]
```diff
@@ -7711,3 +7729,60 @@ definitions:
         description: Datetime at creation of this user's accont
         type: string
         format: date-time
+  ReportingResearcher:
+    description: >
+      Payload for the researcher table in the BigQuery reporting dataset.
+    type: object
+    properties:
+      username:
+        description: Username (GSuite)
```
Comment (jmthibault79) [inline api/src/main/resources/workbench-api.yaml:7738]: Please clarify whether this includes the domain

File: api/src/main/resources/workbench-api.yaml [hunk 2]
```diff
@@ -7711,3 +7729,60 @@ definitions:
         description: Datetime at creation of this user's accont
         type: string
         format: date-time
+  ReportingResearcher:
```
Comment (jmthibault79) [inline api/src/main/resources/workbench-api.yaml:7732]: Which elements are required for each of these structures?  We can add that in a `required` section.

File: api/src/main/java/org/pmiops/workbench/reporting/ReportingService.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package org.pmiops.workbench.reporting;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.pmiops.workbench.model.ReportingSnapshot;
+
+public interface ReportingService {
+  @VisibleForTesting
+  ReportingSnapshot getSnapshot();
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/reporting/ReportingService.java:8]: these are both `public` in the impl

Comment (jaycarlton) [reply api/src/main/java/org/pmiops/workbench/reporting/ReportingService.java:8]: right, It's only in the interface (thus public when overridden) for testing. But this is already changed in the next batch of code.

File: api/src/main/resources/workbench-api.yaml (hunk 2 above)
Comment (jaycarlton) [reply api/src/main/resources/workbench-api.yaml:7732]: This isn't part of the API at all, though all environments will be saddled with it. I want to replace it with Immutables or maybe Kotlin data classes so we don't have to put it here.

File: api/src/main/resources/workbench-api.yaml (hunk 1 above)
Comment (jaycarlton) [reply api/src/main/resources/workbench-api.yaml:7738]: yes it does, though now that I think about it, we'll be documenting the schema separately from this file so maybe I should pull these descriptions from here.

We should probably make a new yaml file for internal, backend-only codegen, but it's not worth it compared to just using Immutables.

**Pull request merged**